statsmodels/discrete/discrete_model.py

```diff
@@ -3157,19 +3136,17 @@ def fit_regularized(self, start_params=None, method='l1',
             alpha=0, trim_mode='auto', auto_trim_tol=0.01, size_trim_tol=1e-4,
             qc_tol=0.03, **kwargs):
 
-        if method not in ['l1', 'l1_cvxopt_cp']:
-            raise TypeError(
```

Comment:
This check is done in DiscreteModel.fit_regularized, so is redundant here.  Also `TypeError` is _not_ the right exception to be raising here.